app/js/question_controller.js

```diff
@@ -13,6 +13,32 @@
   var question_object;
 
   /**
+   * Handles clicks on the helpful vote icon.
+   */
+  function helpful_votes_handler() {
+    var question_thread = document.getElementById('question-thread');
+    question_thread.addEventListener('click', function(event) {
+
+      var elem = event.target;
+
+      if ('vote' === elem.className) {
+        var answer_id = elem.dataset.id;
+        SumoDB.submit_vote(answer_id).then(function(response) {
+          if (response.num_helpful_votes) {
+            elem.textContent = response.num_helpful_votes;
+          } else {
+            // the user has already voted.
+            if (response.message === 'CONFLICT') {
+              // TODO: Show this in a user message dialog
+              console.log('Cannot vote twice for the same answer.');
```

Comment:
I don't think we need to do anything in the case a user has already voted.
In that case, I think this is enough:
```js
} else if (response.message === 'CONFLICT') {
  // Do nothing, user has already voted
}
```